test/TritonGPU/amd/amd-reorder-instructions.mlir

```diff
@@ -792,3 +792,26 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 4 :
     tt.return
   }
 }
+
+// -----
+
+// CHECK-LABEL: sink_convert_dealloc
+//       CHECK: triton_gpu.async_wait {num = 0 : i32}
+//       CHECK: triton_gpu.local_dealloc %0 : !tt.memdesc<4x128x64xf16, #shared, mutable>
+//       CHECK: triton_gpu.local_dealloc %1 : !tt.memdesc<4x128x64xf16, #shared, mutable>
+//       CHECK: %3 = triton_gpu.convert_layout %arg0 : tensor<32x32xf32, #blocked> -> tensor<32x32xf32, #blocked1>
+#blocked = #triton_gpu.blocked<{sizePerThread = [1, 1], threadsPerWarp = [32, 1], warpsPerCTA = [1, 4], order = [0, 1]}>
+#blocked1 = #triton_gpu.blocked<{sizePerThread = [1, 1], threadsPerWarp = [32, 1], warpsPerCTA = [1, 4], order = [1, 0]}>
+#shared = #triton_gpu.shared<{vec = 8, perPhase = 1, maxPhase = 4, order = [0, 1]}>
+module attributes {"triton_gpu.num-warps" = 4 : i32, "triton_gpu.threads-per-warp" = 32 : i32} {
+  tt.func public @sink_convert_dealloc(%arg0: tensor<32x32xf32, #blocked>) attributes {noinline = false} {
+    %0 = triton_gpu.local_alloc : () -> !tt.memdesc<4x128x64xf16, #shared, mutable>
+    %1 = triton_gpu.local_alloc : () -> !tt.memdesc<4x128x64xf16, #shared, mutable>
+    %2 = triton_gpu.convert_layout %arg0 : tensor<32x32xf32, #blocked> -> tensor<32x32xf32, #blocked1>
+    triton_gpu.async_wait {num = 0 : i32}
```

Comment:
Do we need this async_wait in this example?

Reply:
I'm not sure the reusability on this pass particularly given how we order instructions are strongly implied by the GPU architectures--the pipes of different hardware issuing ports and other resources and their relationships to hide. I kinda see divergence even between CDNA and RDNA. So I'll remove this for now.

Reply:
I'm seeing this pass (and the common one) as a collection of different reordering logics. Currently they have only a few heuristics but will be growing later.
I was thinking of having those set of logics together in the pass and enabling them differently by given profile, e.g., `default` profile enables the set of reordering logics to reduce register pressure and `amdgpu` profile enables reordering logics to prefetch data better. 
Or maybe those logics might be diverged to separate passes as they grow.

Reply:
yeah we need to see how this goes.